### ISSUE TITLE
CategoryChannel.edit fails to update overwrites

### ISSUE THREAD

**Issue by cyrus01337** [Rapptz/discord.py#5169]
### Summary

When updating a CategoryChannel's overwrites they are not applied, while locally the children channels were updated and not the category.

### Reproduction Steps

Testing in Jishaku eval and a standard command
```
import discord

# alter accordingly
grotto = discord.utils.find(lambda v: v.name.lower().startswith("grotto"), _ctx.guild.roles)
overwrites = {
    grotto: discord.PermissionOverwrite(send_messages=True),
    _author: discord.PermissionOverwrite(read_message_history=True)
}

for channel in _ctx.guild.categories:
    await channel.edit(overwrites=overwrites)
```
```
@commands.is_owner()
@commands.group(hidden=True)
async def test(self, ctx):
    grotto = discord.utils.find(lambda v: v.name.lower().startswith("grotto"), ctx.guild.roles)
    overwrites = {
        grotto: discord.PermissionOverwrite(send_messages=True),
        ctx.author: discord.PermissionOverwrite(read_message_history=True)
    }

    for channel in ctx.guild.categories:
        await channel.edit(overwrites=overwrites)
    await ctx.message.add_reaction("👍")
```

### Expected Results

All channels (the category in question as well as any channels under it) have their overwrites updated accordingly as per the earlier code snippet.

### Actual Results

This was not achieved, instead the children channels had their overwrites applied while the category remains unchanged. This was unable to be reproduced by other members and instead **none** of the channels were updated.

No traceback available.

### Checklist

- [x] I have searched the open issues for duplicates.
- [x] I have shown the entire traceback, if possible.
- [x] I have removed my token from display, if visible.

### System Information

- Python v3.8.5-final
- discord.py v1.3.4-final
- aiohttp v3.6.2
- websockets v8.1
- system info: Windows 10 10.0.19041

**Comment by XuaTheGrate** [comment]
Using `v1.4.0a2393+g36f3dca`, I'm getting inverted behaviour - where the children are not updated, only the parent categories are.

**Comment by SebbyLaw** [comment]
Using v1.3.4 as well, but children are not updated. Nothing happens to the category as described in the original issue though. Tested on a guild with bot as owner.

**Comment by cyrus01337** [comment]
To clarify, the children actually were *not* updated and this is actually a mishap on my end, however the category permissions issue (being the main issue) still persists. Apologies.

**Comment by cyrus01337** [comment]
This was apparently solved (though the fix is unknown) in v1.4 (1.4.0a2409+gb77af19) and is no longer an issue.
